proto/internal/temporal/server/api/deployment/v1/message.proto

```diff
@@ -27,35 +27,97 @@ package temporal.server.api.deployment.v1;
 option go_package = "go.temporal.io/server/api/deployment/v1;deployment";
 
 import "temporal/api/enums/v1/task_queue.proto";
+import "temporal/api/taskqueue/v1/message.proto";
 import "google/protobuf/timestamp.proto";
 import "temporal/api/deployment/v1/message.proto";
 import "temporal/api/common/v1/message.proto";
 
-// Data for each worker deployment version + task queue pair.
-// This is stored in each worker deployment version (for each task queue),
-// and synced to task queue user data (for each worker deployment version).
-message WorkerDeploymentTaskQueueData {
+// Identifies a Worker Deployment Version. The combination of `deployment_name` and `build_id`
+// serve as the identifier.
+message WorkerDeploymentVersion {
+    // The name of the Deployment this version belongs too.
+    string deployment_name = 1;
+    // Build ID uniquely identifies the Deployment Version within a Deployment, but the same Build
+    // ID can be used in multiple Deployments.
+    string build_id = 2;
 }
 
-// Local state for Worker Deployment Version
-message VersionLocalState {
-    string deployment_name = 1;
-    string version = 2;
+// The source of truth for this data is in the WorkerDeployment entity workflows, which is
+// synced to all TQs whenever the source changes.
+message DeploymentVersionData {
+    WorkerDeploymentVersion version = 1;
+
+    // Last time `current_since_time`, `ramping_since_time, or `ramp_percentage` of this version changed.
+    google.protobuf.Timestamp routing_update_time = 2;
 
-    map<string, TaskQueueFamilyData> task_queue_families = 3;
+    // (-- api-linter: core::0140::prepositions=disabled
+    //     aip.dev/not-precedent: 'Since' captures the field semantics despite being a preposition. --)
+    // Nil if not current.
+    google.protobuf.Timestamp current_since_time = 3;
 
-    bool is_current = 4;
-    google.protobuf.Timestamp last_became_current_time = 5;
+    // (-- api-linter: core::0140::prepositions=disabled
+    //     aip.dev/not-precedent: 'Since' captures the field semantics despite being a preposition. --)
+    // Nil if not ramping. Updated when the version first starts ramping, not on each ramp change.
+    google.protobuf.Timestamp ramping_since_time = 4;
 
-    google.protobuf.Timestamp create_time = 6;
+    // Range: [0, 100]. Must be zero if the version is not ramping (i.e. `ramping_since_time` is nil).
+    // Can be in the range [0, 100] if the version is ramping.
+    float ramp_percentage = 5;
 
-    bool started_deployment_workflow = 7;
+    // Experimental
+    google.protobuf.Timestamp first_poller_time = 6;
```

Comment:
Not urgent, but we can delete this field, it should not be used in the new apis AFAIK. cc @Shivs11 @carlydf

Reply:
deleted in my latest PR, will merge that into here asap